desktop/NodeManager.ts

```diff
@@ -506,14 +609,199 @@ class NodeManager extends AbstractManager {
+      process.stdout.on('data', (data) => {
+        stdout += data;
+      });
+      process.stderr.on('data', (data) => {
+        stderr += data;
+      });
+
+      process.on('error', (err) => {
+        reject(
+          new Error(
+            `Cannot run Quicksync tool to check the database: ${err.message}`
+          )
+        );
+      });
+
+      process.on('close', async (code) => {
+        if (code !== 0) {
+          // TODO: Better error for known codes
+          return reject(
+            new Error(
+              `Quicksync tool failed with exit code ${code}:\n${stderr}`
+            )
+          );
+        }
+
+        const stats = stdout
+          .split('\n')
+          .filter((l) => l.startsWith('Latest') || l.startsWith('Current'))
+          .join('\n')
+          .match(/(\d+)/g);
+
+        if (!stats) {
+          return reject(
+            new Error(
+              `Unknown result of running Quicksync tool:\n${stdout}\n${stderr}`
+            )
+          );
+        }
+
+        const db = parseInt(stats[0], 10);
+        const current = parseInt(stats[1], 10);
+        const available = stats[2] ? parseInt(stats[2], 10) : 0;
+
+        const cfg = await loadNodeConfig();
+        const epochSize = Math.floor(
+          (cfg?.main?.['layers-per-epoch'] || 4032) / 2
+        );
+
+        const result: QuicksyncStatus = {
+          synced: db >= current - epochSize,
+          db,
+          current,
+          available,
+        };
+
+        this.mainWindow.webContents.send(
+          ipcConsts.UPDATE_QUICKSYNC_STATUS,
+          result
+        );
+        return resolve(result);
+      });
+    });
+  };
+
+  private runQuicksyncDownload = async () => {
+    const bin = getQuicksyncPath();
+    const args = [
+      'download',
+      '--node-data',
+      this.getNodeDataPath(),
+      '--go-spacemesh-path',
+      getNodePath(),
+    ];
+
+    logger.log('runQuicksync:download', `${bin} ${args.join(' ')}`);
+
+    return new Promise((resolve, reject) => {
+      const process = spawn(bin, args);
+      let cancelled = false;
+
+      process.on('error', (err) => {
+        reject(
+          new Error(
+            `Cannot run Quicksync tool to download the database: ${err.message}`
+          )
+        );
+      });
+
+      const notifyProgress = (messageLines: string[], isFinish = false) => {
+        showGenericModal(this.mainWindow.webContents, {
+          title: 'Quicksyncing...',
+          message: messageLines.join('\n'),
+          buttons: [
+            isFinish
+              ? {
+                  label: 'Close',
+                  action: 'close',
+                }
+              : {
+                  label: 'Cancel',
+                  action: 'cancel',
+                },
+          ],
+        });
+      };
+
+      ipcMain.once(ipcConsts.GENERIC_MODAL_BTN_PRESS, (_, eventName) => {
+        if (eventName === 'cancel') {
+          cancelled = true;
+          process.kill();
```

Comment:
❗️
1. Start downloading the state, 
1. then cancel (the downloaded db fragment stays in the node-data dir), 
1. then go to settings and start quicksync again - the file that was there continues to be downloaded 
1. but after the quicksync process is complete, we get the :

`2024-02-16T23:08:52.084+0100	FATAL	open sqlite db open db file:/Users/monikasmolarek/Library/Application Support/Spacemesh/node-data/7c8cef2b/state.sql: sqlite.Conn.Prepare: SQLITE_CORRUPT: database disk image is malformed (PRAGMA journal_mode=wal;)	{"name": ""}
`

and of course, if the user really wants to cancel and not use the trusted state, the partially downloaded file would remain in this directory silently, taking several GB (or maybe even more in the future)